### DISCUSSION TITLE
Add some dynamic opt-out for magit-diff-refine-hunk

### DISCUSSION THREAD

**Discussion by darkfeline** [magit/magit#4961] [Ideas]
I think we could add some basic dynamic behavior for `magit-diff-refine-hunk` because it's a great feature but there are some specific cases where its performance tanks hard, especially when dealing with generated files such as protobufs.

For example, it could be temporarily disabled for commits with diffs larger than a specific size, or lines longer than a specific length, or for filenames matching a set of regexps.

**Comment by tarsius** [comment]
Refinement can be toggled using `D t` (`magit-diff-toggle-refine-hunk`). I do not intend you somehow detect that refinement might be problematic and then automatically disable. The plan is to make it asynchronous, so that it does not matter anymore if it occasionally takes a long time.
